src/main/java/com/google/devtools/build/lib/bazel/repository/cache/RepositoryCache.java

```diff
@@ -310,6 +312,15 @@ public static void assertFileChecksum(String expectedChecksum, Path filePath, Ke
    */
   public static String getChecksum(KeyType keyType, Path path)
       throws IOException, InterruptedException {
+    // Attempt to use the fast digest if the hash function of the filesystem
+    // matches `keyType` and it's available.
+    if (path.getFileSystem().getDigestFunction().getHashFunction().equals(keyType.hashFunction)) {
+      byte[] digest = path.getFastDigest();
+      if (digest != null) {
+        return BaseEncoding.base16().lowerCase().encode(digest);
```

Comment:
question -- why does this need to be base64 encoded? The "normally computed" checksum below doesn't seem to be base64 encoded.

Reply:
ah sorry, I saw "base**" and immediately assumed it was 64... this is completely correct :)